File: lib/data/emv_result.dart
// ignore: file_names
part of '../microatmsdk.dart';

class EmvResult {
  late String communication;
  late String cardType;
  late String pan;
  late String t2d;
  late String icCardData;
  late String expDate;

  EmvResult({
    this.communication = "",
    this.cardType = "",
    this.pan = "",
    this.t2d = "",
    this.icCardData = "",
    this.expDate = "",
  });
  EmvResult.fromJson(Map<String, dynamic> json) {
    communication = json['communication'];
    cardType = json['cardType'];
    pan = json['pan'];
    t2d = json['t2d'];
    icCardData = json['icCardData'];
    expDate = json['expDate'];
  }
  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['communication'] = communication;
    data['cardType'] = cardType;
    data['pan'] = pan;
    data['t2d'] = t2d;
    data['icCardData'] = icCardData;
    data['expDate'] = expDate;
    return data;
  }
}

File: android/src/main/kotlin/id/bni46/microatmsdk/data/EmvResult.kt
package id.bni46.microatmsdk.data

import androidx.annotation.Keep
import com.google.gson.Gson

@Keep
data class EmvResult(
    val communication: String,
    val cardType: String,
    val pan: String,
    val t2d: String,
    val icCardData: String,
    val expDate: String
) {
    fun toJson(): String? {
        return Gson().toJson(this)
    }
}